### ISSUE TITLE
fix: `Failed to run serverpod. You need to have dart installed and in your $PATH`

### ISSUE THREAD

**Issue by tomassasovsky** [serverpod/serverpod#249]
On Windows, this is the check that's being run to verify that Dart is indeed present in the PATH:

https://github.com/serverpod/serverpod/blob/e529b9333ce6057fd53d16d86597c14c34bc142c/tools/serverpod_cli/bin/serverpod_cli.dart#L44

This will cause an error, as the file is not named just `dart`, but `dart.exe`

**Comment by vlidholt** [comment]
Which version of Windows are you running?

**Comment by tomassasovsky** [comment]
Windows 11, Version 10.0.22000 Build 22000.

**Comment by vlidholt** [comment]
Thanks! Will try to fix in version 0.9.11

**Comment by yahu1031** [comment]
@tomassasovsky Looks like both works.
![image](https://user-images.githubusercontent.com/35523357/187689855-3962b1f2-341f-4f74-bec7-1a6d917c7e57.png)
### My Sys Configs
```console
Device name	Minnu
Processor	AMD Ryzen 7 2700X Eight-Core Processor            3.70 GHz
Installed RAM	16.0 GB
Device ID	3F583939-41AA-4002-BBCE-E80160D264B6
Product ID	00330-80000-00000-AA902
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display
Edition	Windows 11 Pro Insider Preview - Version 22H2
Installed on	‎19-‎08-‎2022
OS build	25182.1000
Experience	Windows Feature Experience Pack 1000.25182.1000.0

```

Can you please cross-check that with a sample test file as mentioned below and LMK

`tools/serverpod_cli/test/main.dart`

```dart
import 'package:test/test.dart';

import '../bin/util/command_line_tools.dart';

void main() {
  group('CommandLineTools test', () {
    test('for dart existsCommand', () async {
      var dartExists = await CommandLineTools.existsCommand('dart');
      expect(dartExists, true);
    });
    test('for dart.exe existsCommand', () async {
      var dartExeExists = await CommandLineTools.existsCommand('dart.exe');
      expect(dartExeExists, true);
    });
  });
}
```

**Comment by sanjeevbishnoi** [comment]
Same error occurs when someone is using FVM for flutter. When flutter and dart is installed using fvm then the installation is done at $HOME->fvm->stable->bin  (in case you are not using stable ,then replace stable with that version name, you will find it when you browse it ) on Mac. 
if we run 'which' command it returns empty in terminal. it will print nothing. 

then only one thing to do, you need to enter the path by following this guide [update path]( https://docs.flutter.dev/get-started/install/macos#update-your-path)

in my case it was `$HOME/.bash_profile` just open it using vim
`vim $HOME/.bash_profile`
 and add a path like this. 
`export PATH=$HOME/fvm/versions/stable/bin:$PATH`
replace 'stable' in case you are using different version. 

<img width="583" alt="Screenshot 2022-12-26 at 8 09 15 AM" src="https://user-images.githubusercontent.com/34705304/209493977-6d718d76-9bca-4404-9fbd-c608fc5908c4.png">

close the terminal and reopen it. then run command 'serverpod' it will always work.Thanks @vlidholt for great framework.

**Comment by yahu1031** [comment]
@sanjeevbishnoi Can you check `which dart` and see if you get any output?

**Comment by sanjeevbishnoi** [comment]
@yahu1031  echo $PATH and make sure it contains the highlighted area, these things may vary on mac m1 series and intel series.  Main idea is to provide serverpod access to dart and flutter which is installed using FVM.  
<img width="1434" alt="Screenshot 2022-12-26 at 5 00 24 PM" src="https://user-images.githubusercontent.com/34705304/209544209-8a7bb0c2-090b-46a8-96ac-d770783a1e2a.png">

**Comment by yahu1031** [comment]
@sanjeevbishnoi Okie, could you please share the error too?

**Comment by sanjeevbishnoi** [comment]
Error : Failed to run serverpod. You need to have dart installed and in your $PATH

**Comment by yahu1031** [comment]
@sanjeevbishnoi will check this. Please open this as a new ticket. I will follow it up.

**Comment by sanjeevbishnoi** [comment]
@yahu1031  after following the steps mentioned in my first comment I was solved. Now everything is working. Only thing that is surprising is that when I was typing flutter or dart in the terminal the since last three years It was working. But when I typed 'serverpod' it did not pick the same dart and flutter which was being picked by terminal. I browsed the path of dart and flutter excutables from folder generated by fvm and inserted into PATH and restarted the terminal and everything worked. It is one of the weird bug which can not be replicated on same system again. A user has already  have pull request **Check fvm installed #488** for same bug.
